src/screens/BackupWords.js

```diff
@@ -0,0 +1,241 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import _ from 'lodash';
+import {
+  Image,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native';
+import NewHathorButton from '../components/NewHathorButton';
+import HathorHeader from '../components/HathorHeader';
+import FeedbackModal from '../components/FeedbackModal';
+import checkIcon from '../assets/images/icCheckBig.png';
+import errorIcon from '../assets/images/icErrorBig.png';
+
+import baseStyle from '../styles/init';
+
+import { HATHOR_COLOR } from '../constants';
+import { Strong } from '../utils';
+
+
+class BackupWords extends React.Component {
+  /**
+   * step {number} Which validation step user is
+   * indexes {Array} Array of indexes that will be used to execute the validation
+   * wordsOptions {Array} For each step we have some options for the user to choose the correct word
+   * modal {FeedbackModal} modal to display. If null, do not display
+   */
+  state = {
+    step: 0,
+    indexes: [],
+    wordOptions: [],
+    modal: null,
+  };
+
+  style = Object.assign({}, baseStyle, StyleSheet.create({
+    footerView: {
+      width: 8,
+      height: 8,
+      borderRadius: 4,
+      marginRight: 8,
+    },
+    past: {
+      backgroundColor: '#000',
+    },
+    current: {
+      backgroundColor: HATHOR_COLOR,
+    },
+    future: {
+      backgroundColor: '#000',
+      opacity: 0.3,
+    },
+    lastView: {
+      marginRight: 0,
+    },
+    button: {
+      marginBottom: 16,
+    },
+  }));
+
+  // Array of words shown on the previous screen
+  words = null;
+
+  /**
+   * Expects 'words' as navigation parameter with all generated words as a string separated by space
+   */
+  constructor(props) {
+    super(props);
+    const paramWords = this.props.navigation.getParam('words');
+    this.words = paramWords ? paramWords.split(' ') : [];
+  }
+
+  componentDidMount() {
+    // Get 5 random indexes of word to check backup
+    const indexesArr = Array(24).fill().map((v, i) => i + 1);
+    const indexesToBackup = _.shuffle(indexesArr).slice(0, 5);
+    this.setState({ indexes: indexesToBackup }, () => {
+      this.updateWordOptions();
+    });
+  }
+
+  /**
+   * Update state with options to be shown to the user depending on the step
+   * If the step is the word in position 4, we must show the words in position 2, 3, 4, 5, 6
+   * If we get one of the corner cases
+   * (positions 1, 2, 23, or 24 we expand to the side we still have words) e.g., if we want to
+   * validate word in position 2, we must have 5 options.
+   * Positions 1, 2, 3, 4, 5 (the position 0 is substituted by position 5)
+   */
+  updateWordOptions = () => {
```

Comment:
Maybe rename it to `updateWordsShownOnScreen`.

Reply:
Done